### ISSUE TITLE
Add ShortSaleRestriction to QuickFix/J dictionary 

### ISSUE THREAD

**Issue by david-gibbs-ig** [quickfix-j/quickfixj#267]
Tag 1687 ShortSaleRestriction  was added added in  FIX.5.0SP2  EP120

https://fiximate.fixtrading.org/en/FIX.5.0SP2_EP254/tag1687.html

We would like to use it so will add to :
quickfixj-messages/quickfixj-messages-fix50sp2/src/main/resources/FIX50SP2.modified.xml

**Comment by chrjohn** [comment]
Hmm, I don't know how to go about this. Actually the dictionaries packaged with QFJ are FIX5.0SP2. That's why there are no changes of the EPs incorporated.
We have two possibilities:
* let users handle the EPs by themselves and have them edit the DDs and generate the code by themselves.
* or provide the latest EP dictionary with QFJ. But I would rather have this separate from the SP2 dictionary since it is confusing otherwise.

Thoughts? Maybe we should have a poll on the mailing list.

**Comment by david-gibbs-ig** [comment]
In future I think the way to do this is to use the FIX Orrchestra project tools to generate the QFJ dictionary from an "orchestration" and to change the QFJ build to accept the QFC dictionary as a dependency. For now my suggestion is that ,since any official EP, as documented in Fiximate is additive and non-conflicting, we be free to add them on top of the baseline. I think that adding the EPs in derived projects isn't ideal. FTC are talking about moving to "latest" but I don't understand how this will work. Perhaps 50SP2-latest is a good workaround for now ?

**Comment by david-gibbs-ig** [comment]
FTC are going to move to a "latest" release. This will be immutable versioned with the latest EP. I think maybe we can add "latest" to QFJ and leave the legacy versions alone. That should line up with what FTC are doing. I've suggested that QFJ can get latest as a dependency from an FTC built QFJ dictionary that will be produced when "latest" is updated. This FTC artifact does not yet exist to be published but if you are open to this then this would provide incentive to generate and publish it.

**Comment by chrjohn** [comment]
I agree we should add "latest" to QFJ. We only need to think about how best to do it, e.g. need to check if there are any hardcoded assumptions in the code. This would probably also mean that "latest" would replace the "fix50sp2" packages. I am not sure how that would work out in the maven build. Or would it have to be called "fix-latest"? But it would still be FIX50SP2 on the wire, right?
If FTC/Orchestra could build the latest dictionary that would be great. I was thinking about doing all the dictionary generations with the Orchestra provided tool. Also see https://github.com/quickfix-j/quickfixj/issues/258

**Comment by david-gibbs-ig** [comment]
I have tried building the QFJ Dictionary for Orchestra, it build a QuickFIX dictionary OK but there will be a little post-processing required as there are some things like case in redundant enumerations that cause issues with the QFJ build. I have not revisited it yet to get it working. We can get a call with the Orchestra group to discuss so that there can be a "supported" process. I have proposed this and they intend to organise a call with "vendors". I think that the dictionaries could be generated artifacts that are published by another build thus giving stability to the QFJ build. The QFJ build would then depend  on versioned published artifacts. The published artifacts for "latest" will also be immutable and consecutive releases will be versioned using the EP number or similar. In the bigger picture what would be great would be to make it very easy to extend (and/or reduce) the QFJ build to produce custom builds simply by changing some dependency or dependencies. The custom dictionaries would be built and published by the party with a custom Rules of Engagement. A custom QFJ build could be likewise built.
